docs/guides/primitive-input-output.ipynb

```diff
@@ -0,0 +1,284 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "e8a8a4b1-2ed2-4b82-8d02-98174a6da903",
+   "metadata": {},
+   "source": [
+    "# Understanding primitive inputs and outputs\n",
```

Comment:
should be "understand" not "understanding" (if @abbycross has taught me anything in the last year 😆 ). but i think we could probably just drop the verb entirely.
```suggestion
    "# Primitive inputs and outputs\n",
```

Reply:
Agreed - I meant to comment on that actually :D